### PR TITLE
[2.9] Show Cloud Credential Expiration information for Harvester Creds, add actions to renew

### LINKED ISSUE #11270: [backport v2.9.x] Allow users to easily see expiring harvester cloud credentials and to renew them
Epic: https://github.com/rancher/dashboard/issues/11256 SURE-8516

Requires expiring label on cloud cred

- Global notification somewhere warning of expired / soon to expire harv cloud creds
- Clearly see in cloud cred table expired / soon to expire (see cluster dashboard certs list for pattern?)
  - Column for sorting, only if harv cloud cred present?
- Sub row in harv clusters list to warn that cluster has a cloud cred that's expiring
- Cloud Cred bulk action to renew (recreate kubeconfig, update cloud cred)

Edit: Blocked on https://github.com/rancher/rancher/issues/47243

### PR DESCRIPTION
**Pull request by richard-cox**
### Summary
Fixes #11270

### Occurred changes and/or fixed issues
- Cloud Credentials List
  - If there's a cloud cred of type harvester,  show an `Expires` column
  - Expires column is populated ig cloud cred has expired annotation (empty if not)
  - For cloud creds with an expiration, show a 'Renew' action
  - Renew action is provider specific, provide one for harvester which updates the kubeconfig within the cred
- Cluster List
  - If there's a cloud cred associated with the harvester cluster (cloud cred `harvestercredentialConfig.clusterId`) check it's expired date, warn if within 7 days of expiring or has expired
  - If there's a cloud cred with expiration, show the `Renew Cloud Credential` action
  - Action does same as cloud cred renew
- Banners
  - In both lists show a banner warning of soon to expire or expired creds

### Areas or cases that should be tested
- End to end
  -  Renew Cloud Credential results in cluster continuing to function

### Areas which could experience regressions
- Cloud Credentials List
- Prov Cluster List

#### Cloud Credentials List

image

image

#### Clusters List

image

image

### Checklist
- [x] The PR is linked to an issue and the linked issue has a Milestone, or no issue is needed
- [x] The PR has a Milestone 
- [x] The PR template has been filled out
- [x] The PR has been self reviewed 
- [x] The PR has a reviewer assigned
- [x] The PR has automated tests or clear instructions for manual tests and the linked issue has appropriate QA labels, or tests are not needed
- [x] The PR has reviewed with UX and tested in light and dark mode, or there are no UX changes

### CODE REVIEW

File: shell/config/labels-annotations.js [hunk 1]
```diff
@@ -172,3 +172,7 @@ export const SYSTEM_LABELS = [
   'node.kubernetes.io',
   'egress.rke2.io'
 ];
+
+export const CLOUD_CREDENTIALS = [
+  'cattle.io/expiryTime'
```
Comment (richard-cox) [inline shell/config/labels-annotations.js:177]: now `cattle.io/expiration-date`

File: shell/config/labels-annotations.js [hunk 2]
```diff
@@ -172,3 +172,7 @@ export const SYSTEM_LABELS = [
   'node.kubernetes.io',
   'egress.rke2.io'
 ];
+
+export const CLOUD_CREDENTIALS = [
+  'cattle.io/expiration-date'
```
Comment (torchiaf) [inline shell/config/labels-annotations.js:177]: This is `cattle.io/expiration-timestamp` in rancher 2-9-head

**Review by torchiaf — CHANGES_REQUESTED** [review]

**Review by torchiaf — APPROVED** [review]
UI changes look good.
However, the Renew actions fail because the `cattle.io/expiration-timestamp` label in cloud credentials object is not updated by the backend.

**Comment by richard-cox** [comment]
The `Renew` issue was investigated today and to progress it @MbolotSuse is taking a look. Given timing we'll merge this PR WITH the feature, and then hide / change messaging tomorrow if there's no solution

**Pull request merged**